runtime/src/main/java/com/redhat/ipaas/rest/ComponentGroups.java

```diff
@@ -37,12 +38,15 @@
     @Inject
     private DataManager dataMgr;
 
+    @Context
+    private UriInfo uri;
+
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "List component groups")
     @ApiResponses(value = {@ApiResponse(code = 200, message = "Success", response = ComponentGroup.class)})
     public Collection<ComponentGroup> list() {
-        return dataMgr.fetchAll(ComponentGroup.class);
+        return dataMgr.fetchAll(ComponentGroup.class, new ReflectiveSorter<>(ComponentGroup.class, uri.getQueryParameters()));
```

Comment:
But then again you would either build up the `SortOptions` by specifying both parameters `sortField` and `sortDir` as parameters so that each fetchAll() method would need to extract it by themselve or you would couple `SortOptions` also to `MultiValueMap`.

The only solution I would see it to copy over the whole MultiValueMap into a regular Map (but the losing multi-values which might help when introducing multiple sortFields for multiple sort order) or put it behind a facade (which might be the simpler solution). Let me try the latter.